datafusion/core/src/datasource/listing/table.rs

```diff
@@ -797,7 +787,7 @@ impl TableProvider for ListingTable {
             // all of the data at the input is sink after execution finishes. See discussion for rationale:
             // https://github.com/apache/arrow-datafusion/pull/7610#issuecomment-1728979918
             unbounded_input: is_plan_streaming(&input)?,
-            single_file_output: self.options.single_file,
+            single_file_output: false,
```

Comment:
this got me thinking we can remove the option from the FileSInkConfig itself. 

However I see how that this PR is removing the `single_file` option from the *read* side (the ListingTable) but it can also potentially be important for the write side (aka a COPY statement)

Reply:
Yes, COPY still uses single_file_output option of the FileSinkConfig struct. We could potentially update COPY to instead rely on inference based on the path rather than an explicit option. E.g. copy table to file.parquet vs copy table to folder/. Then single_file_output could be removed entirely.

Reply:
🤔  that certainly would be a better UX in my opinion (figuring out to use SINGLE_FILE  in https://github.com/apache/arrow-datafusion/blob/98a5a4eb1ea1277f5fe001e1c7602b37592452f1/datafusion/sqllogictest/test_files/repartition_scan.slt#L35-L38 was actually quite painful for me)

Shall I propose a ticket to do it?

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/8621